lib/wings/model_transformer.rb

```diff
@@ -196,11 +196,37 @@ def internal_resource
 
     class AttributeTransformer
       def self.run(obj, keys)
-        keys.each_with_object({}) do |attr_name, mem|
-          next unless obj.respond_to? attr_name
+        # TODO: There is an open question about whether we want to treat all these relationships the same.  See Issue #3904.
+        attrs = keys.select { |k| k.to_s.end_with? '_ids' }.each_with_object({}) do |attr_name, mem|
+          mem[attr_name.to_sym] =
+            TransformerValueMapper.for(obj.try(attr_name)).result ||
+            TransformerValueMapper.for(attr_ids_for_name_minus_ids(obj, attr_name)).result ||
+            TransformerValueMapper.for(attr_ids_for_name_minus_ids_pluralized(obj, attr_name)).result || []
+        end
+        keys.each_with_object(attrs) do |attr_name, mem|
+          next unless obj.respond_to?(attr_name) && !mem.key?(attr_name.to_sym)
           mem[attr_name.to_sym] = TransformerValueMapper.for(obj.public_send(attr_name)).result
         end
       end
+
+      def self.attr_ids_for_name_minus_ids(obj, attr_name)
```

Comment:
i'm finding these method names hard to read, and noticing that most of the method code is duplicated.

what do you think about a refactor like:

```ruby
# ...
        attrs = keys.select { |k| k.to_s.end_with? '_ids' }.each_with_object({}) do |attr_name, mem|
          mem[attr_name.to_sym] =
            TransformerValueMapper.for(obj.try(attr_name)).result ||
            TransformerValueMapper.for(attribute_ids_for(name: attr_name.chomp('_ids'), obj: obj)).result ||
            TransformerValueMapper.for(attribute_ids_for(name: attr_name.chomp('_ids').pluralize, obj: obj)).result || []
        end
# ...
def attribute_ids_for(name:, obj:)
  attribute_value = obj.try(name)
  Array(attribute_value).map(&:id) # i think this is equivalent to L216-217 and L222-223
end
```

Reply:
keeping `#attr_name_minus_ids` as a separate method also seems like a good option (even if it can be refactored to just `chomp('_ids')`). but if we keep it, renaming to something that scans a little easier would be helpful. something like  `#attribute_name_without_ids` or `#drop_ids_from_name(name)`/`#drop_ids_from(name:)`?

Reply:
I like it!